doc/tutorial/search/index.md

```diff
@@ -0,0 +1,257 @@
+However, it also needs to store additional information to be bidirectional. You can think of the Bi-FM-Index as two FM-Indices, one for each direction, hence you need about 2 times more space for the Bi-FM-Index.<br>
+
+## Example
+
+Both FM- and Bi-FM-Indices internally use the implementation of the [Succinct Data Structure Library](https://github.com/simongog/sdsl-lite).<br>
+Constructing a (Bi-)FM-Index is very simple:
+
+\include doc/tutorial/search/basic_index.cpp
+
+You can also index text collections (e.g., genomes with multiple chromosomes or protein databases):
+
+\snippet doc/tutorial/search/small_snippets.cpp text_collection
+
+The indices can also be stored and loaded from disk.
+Both seqan3::fm_index::store and seqan3::fm_index::load return a `bool` indicating whether the operation was successful.
+
+\snippet doc/tutorial/search/small_snippets.cpp store
+
+\snippet doc/tutorial/search/small_snippets.cpp load
+
+Note that in contrast to the construction via a given `text`, the template cannot be deduced by the compiler on default construction so you have to provide template arguments.
+
+\assignment{Assignment 1}
+You are given the text
+\code
+dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+\endcode
+Create an fm_index over the reference, store the index and load the index into a new fm_index object.
+Print whether the store and load operations were successful.
+\endassignment
+
+\solution
+\snippet doc/tutorial/search/solution1.cpp solution
+**Expected output:**
+~~~console
+Index stored successfully.
+Index loaded successfully.
+~~~
+\endsolution
+
+# Search
+
+Using an index, we can now conduct searches for a given query. In this part we will learn, how to search exactly, allow substitutions and indels, and how to configure what kind of results we want, e.g. all results vs. only the best result.
+
+## Terminology
+### Search scheme
+A search scheme is the manner of how a search is conducted, e.g. trying to match each position in the reference to the query.
+### Optimal Search Scheme
+The scheme that requires the least amount of queries in the index is called an *optimal search scheme*.<br>
+SeqAn3 provides *optimal* search schemes for up to **4** errors for Bi-FM-Indices and provides sub-optimal, but still efficient search schemes otherwise.<br>
+You can read more about search schemes in [this paper](http://doi.org/10.1101/301085).
+
+\note The choice of the search scheme is done internally you don't have to worry about it. You can only choose between the FM and Bi-FM-Index but when using >= 2 errors it is advisable to use a Bi-FM-Index since searches will be faster.<br>
```

Comment:
Please do not mention things in the tutorial that readers don't have to worry about? Otherwise why would you bother explaining to them what it is?

Reply:
Just say that for more than 2 errors using the Bi-FM-Index is more efficient.